### PR TITLE
Add consumer profiles

### PR DESCRIPTION
**Pull request by dmathieu**
#### Description
This adds profiles support for consumers.

#### Link to tracking issue
Based on the discussion in #10375.

### CODE REVIEW

**Comment by dmathieu** [comment]
I believe the failures is unrelated to these changes.

File: consumer/consumer.go
Comment (mx-psi) [inline consumer/consumer.go:12]: Can you add a docstring here? (Same as the one in the internal package)

File: consumer/consumer.go
Comment (mx-psi) [inline consumer/consumer.go:16]: ditto

File: .chloggen/profile-consumer.yaml
Comment (mx-psi) [inline .chloggen/profile-consumer.yaml:4]:
```suggestion
change_type: new_component
```

File: .chloggen/profile-consumer.yaml
Comment (mx-psi) [inline .chloggen/profile-consumer.yaml:7]:
```suggestion
component: consumer/consumerprofiles
```

File: consumer/consumerprofiles/profiles.go
Comment (mx-psi) [inline consumer/consumerprofiles/profiles.go:1]: This needs a new `go.mod` file so that we can mark `consumer` 1.0 while keep `consumer/consumerprofiles` 0.x

File: consumer/consumerprofiles/go.mod
Comment (mx-psi) [inline consumer/consumerprofiles/go.mod:13]: Why not just? 
```suggestion
	go.opentelemetry.io/collector/consumer v0.104.0
```

**Review by mx-psi — APPROVED** [review]
LGTM. Since this is the first PR actually adding implementation for profiles after the RFC, I am going to wait a few days to let other @open-telemetry/collector-approvers chime in

For future modules, it would be great to split changes of internal-only changes and new modules into separate PRs

**Comment by mx-psi** [comment]
@open-telemetry/collector-approvers planning to merge this tomorrow if nobody blocks before then

**Pull request merged**
